examples/light-switch-app/qpg/src/AppTask.cpp

```diff
@@ -93,6 +94,9 @@ void AppTask::InitServer(intptr_t arg)
     {
         PlatformMgr().ScheduleWork(OpenCommissioning, 0);
     }
+
+    auto & logProvider = LogProvider::GetInstance();
+    chip::app::Clusters::DiagnosticLogs::SetDelegate(kRootEndpointId, &logProvider);
```

Comment:
same here: we should aim at minimal API disruption.

It may also happen that we cannot preserve the API, in which case we should update https://github.com/project-chip/connectedhomeip/blob/master/docs/upgrading.md